changed collegue smart pointer

AntoniaSzecsi opened the pull request.

AntoniaSzecsi committed:
changed collegue smart pointer

CristianIonutIonescu commented on marc_pop_unique_ptr/custom_unique_ptr.h:
```diff
@@ -2,9 +2,20 @@
+#include <utility>
+
+// Here I ran into a big problem for using the standard deleter that Marc used, when trying to push_back the unique pointer into a vector of unique pointers
+// I was getting an error that the copy constructor was deleted. This was because the standard deleter was not copyable and the unique pointer was trying to copy the deleter,
```
unique_ptr shouldn't be copiable at all. If the default deleter was preventing you from copying than it was doing the right thing, try to go back to the previous implementation and debug why it's happening.

AntoniaSzecsi committed:
fix(homework-6): warnings and PR comm

I understood my mistake with the unique ptr. after a bit more debugging I found out the conversion from Dervided classes to Base classes in my STL container was the problem
